plugins/auth-backend/src/providers/aws-alb/provider.ts

```diff
@@ -25,73 +27,174 @@ import { KeyObject } from 'crypto';
 
-  constructor(
-    logger: Logger,
-    catalogClient: CatalogApi,
-    options: AwsAlbAuthProviderOptions,
-  ) {
-    this.logger = logger;
-    this.catalogClient = catalogClient;
-    this.options = options;
+  constructor(options: Options) {
+    this.region = options.region;
+    this.issuer = options.issuer;
+    this.authHandler = options.authHandler;
+    this.signInResolver = options.signInResolver;
+    this.tokenIssuer = options.tokenIssuer;
+    this.catalogIdentityClient = options.catalogIdentityClient;
+    this.logger = options.logger;
     this.keyCache = new NodeCache({ stdTTL: 3600 });
   }
+
   frameHandler(): Promise<void> {
     return Promise.resolve(undefined);
   }
 
   async refresh(req: express.Request, res: express.Response): Promise<void> {
+    try {
+      const result = await this.getResult(req);
+      const response = await this.handleResult(result);
+      res.json(response);
+    } catch (e) {
+      this.logger.error('Exception occurred during AWS ALB token refresh', e);
+      res.status(401);
+      res.end();
+    }
+  }
+
+  start(): Promise<void> {
+    return Promise.resolve(undefined);
+  }
+
+  private async getResult(req: express.Request): Promise<AwsAlbResult> {
     const jwt = req.header(ALB_JWT_HEADER);
-    if (jwt !== undefined) {
+    const accessToken = req.header(ALB_ACCESSTOKEN_HEADER);
+
+    if (jwt !== undefined && accessToken !== undefined) {
       try {
         const headers = getJWTHeaders(jwt);
         const key = await this.getKey(headers.kid);
-        const payload = JWT.verify(jwt, key);
+        const claims = JWT.verify(jwt, key) as AwsAlbClaims;
 
-        if (this.options.issuer && headers.iss !== this.options.issuer) {
-          throw new Error('issuer mismatch on JWT');
+        if (this.issuer && headers.iss !== this.issuer) {
+          throw new Error('Issuer mismatch on JWT');
         }
 
-        const resolvedEntity = await this.options.identityResolutionCallback(
-          payload,
-          this.catalogClient,
-        );
-        res.json(resolvedEntity);
+        const fullProfile: PassportProfile = {
+          provider: 'unknown',
+          id: claims.sub,
+          displayName: claims.name,
+          username: claims.email.split('@')[0].toLowerCase(),
+          name: {
+            familyName: claims.family_name,
+            givenName: claims.given_name,
+          },
+          emails: [{ value: claims.email.toLowerCase() }],
+          photos: [{ value: claims.picture }],
+        };
+
+        return {
+          fullProfile,
+          headers,
+          accessToken,
+        };
       } catch (e) {
-        this.logger.error('exception occurred during JWT processing', e);
-        res.status(401);
-        res.end();
+        throw new Error(`Exception occurred during JWT processing: ${e}`);
       }
     } else {
-      res.status(401);
-      res.end();
+      throw new Error('Missing AWS ALB headers');
```

Comment:
Could again use `AuthenticationError` here and above, and perhaps say exactly which headers where missing as can imagine that helping people quite a bit